mpl_interactions/controller.py

```diff
@@ -274,23 +274,28 @@ def save_animation(
         slider = self.controls[param]
         ipywidgets_slider = False
         if "Box" in str(slider.__class__):
-            ipywidgets_slider = True
             for obj in slider.children:
                 if "Slider" in str(obj.__class__):
                     slider = obj
-            N = int((slider.max - slider.min) / slider.step)
+
+        if isinstance(slider, mSlider):
+            min_ = slider.valmin
+            max_ = slider.valmax
+            step = slider.valstep
+        elif "Slider" in str(slider.__class__):
+            ipywidgets_slider = True
             min_ = slider.min
             max_ = slider.max
             step = slider.step
-        elif isinstance(slider, mSlider):
-            min_ = slider.valmin
-            max_ = slider.valmax
-            if slider.valstep is None:
-                N = N_frames if N_frames else 200
-                step = (max_ - min_) / N
-            else:
-                N = int((max_ - min_) / slider.valstep)
-                step = slider.valstep
+        else:
+            raise NotImplementedError(
+                "Cannot save animation for slider of type %s".format(slider.__class__.__name__)
+            )
+
+        if step is None:
+            N = N_frames if N_frames else 200
```

Comment:
Step needs to be defined for the animation to work - see line 301
```suggestion
            N = N_frames if N_frames else 200
            step = (max_ - min_) / N
```

Reply:
Brrr.  I got too used to linters that immediately point out these things... :sweat_smile:

Reply:
You've convinced me to add one! Do you have thoughts on `flake8` vs `pylint` vs `pylama` vs who knows what else?

Reply:
I like flake8 best because it's faster than pylint and it's easy to add plugins (like an [f-string check](https://pypi.org/project/flake8-use-fstring)). pylint is a good additional check, but a bit slow. I'm not familiar with pylama.

I guess you (also) use vscode :P
https://github.com/ianhi/mpl-interactions/blob/5fa461359b1ab49a686cbc8d1c4e27a073454810/.vscode/settings.json#L4

So it's easy to enforce these checks directly in the gui.